Lesson34/Merve/Lesson33/props-drilling-app/src/modules/user/UserDetailsForm/index.jsx

```diff
@@ -0,0 +1,33 @@
+import { useState, useContext } from "react";
+import { UserDispatchContext } from "../UserProvider";
+import './styles.css';
+
+export const UserDetailsForm = ({ username, onNameChange }) => {
+  const [newName, setNewName] = useState(username);
+  const setUser = useContext(UserDispatchContext);
+
+  const handleChange = (e) => {
+    setNewName(e.target.value);
```

Comment:
It would be better to trim and validate the value f the input so only value in the supported format can be used as a new user name.